stage0/src/asm/boot.s

```diff
@@ -130,25 +179,28 @@ _protected_mode_start:
     # Note that this sets the encrypted bit for _all_ entries in the page tables, even
     # if they are unused. This is fine, as those entries will still not have the PRESENT
     # bit set and thus will be ignored by the CPU.
-    mov $pml4_addr, %eax      # Base values for all page tables.
-    mov $pdpt_addr, %ebx
-    mov $pd_addr, %ecx
-    mov $pt_addr, %edx
+    mov ${pml4}, %eax         # Base values for all page tables.
+    mov ${pdpt}, %ebx
+    mov ${pd_0}, %ecx
+    mov ${pd_3}, %edx
+    mov ${pt_0}, %ebp
```

Comment:
I think now we're truly using all of the 32-bit registers for this loop.

Reply:
Indeed. :)

But I plan to stop using ebp  in a follow-up PR when I move creation of pt_0 to early Rust code.